### PR TITLE
API-838: Add user role/group permissions to Connection settings

### PR DESCRIPTION
**Pull request by pchasle**
**Description (for Contributor and Core Developer)**

image

**Definition Of Done (for Core Developer only)**

| Q                                 | A
| --------------------------------- | ---
| Added Specs                       | Todo
| Added legacy Behats               | -
| Added acceptance tests            | Todo
| Added integration tests           | Todo
| Changelog updated                 | -
| Review and 2 GTM                  | Todo
| Micro Demo to the PO (Story only) | Todo
| Migration script                  | -
| Tech Doc                          | -

`Todo`: Pending / Work in progress
`OK`: Done / Validated
`-`: Not needed

### CODE REVIEW

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php [hunk 1]
```diff
@@ -53,6 +61,8 @@ public function __construct(
         $this->image = $image;
         $this->username = $username;
         $this->password = $password;
+        $this->userRoleId = $userRole;
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php]: Is it a code or an id? It's defined as a string

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php [hunk 2]
```diff
@@ -35,15 +35,23 @@ class ConnectionWithCredentials
...
     public function __construct(
         string $code,
         string $label,
         string $flowType,
         string $clientId,
         string $secret,
         string $username,
-        ?string $password = null,
-        ?string $image = null
+        ?string $password,
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php]: I would remove password from the constructor as it is never known from the hydration with your rework.
We often adds optional/nullable parameters at the end of the parameters list.

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/services.yml [hunk 3]
```diff
@@ -29,12 +29,20 @@ services:
             - '@fos_oauth_server.client_manager.default'
             - '@database_connection'
 
-    akeneo_connectivity.connection.service.client.regenerate_user_password:
+    akeneo_connectivity.connection.service.user.regenerate_user_password:
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Symfony/Resources/config/services.yml]: :+1:

File: src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php (hunk 1 above)
Comment (pchasle) [reply src/Akeneo/Connectivity/Connection/back/Domain/Settings/Model/Read/ConnectionWithCredentials.php]: An ID, I will fix the variable name to be `userRoleId`

File: src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/UpdateConnectionEndToEnd.php [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+        ];
+
+        $this->createAdminUser();
+        $this->authenticate('admin', 'admin');
+        $this->client->request('POST', '/rest/connections/franklin', [], [], [], json_encode($data));
+
+        Assert::assertEquals(Response::HTTP_NO_CONTENT, $this->client->getResponse()->getStatusCode());
+    }
+
+    public function test_it_fails_to_update_connection_with_bad_request()
```
Comment (fitn) [inline src/Akeneo/Connectivity/Connection/back/tests/EndToEnd/Connection/UpdateConnectionEndToEnd.php]: Not sure it should be a endToEnd

File: src/Akeneo/Connectivity/Connection/back/tests/Acceptance/Context/ConnectionContext.php [hunk 5]
```diff
@@ -119,8 +119,18 @@ public function iChangeTheOfTheConnectionBy(string $label, TableNode $table)
         $newFlowType = $data['flow_type'];
         $newImage = $data['image'] ?? null;
 
+        // $data['user_role'];
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Acceptance/Context/ConnectionContext.php]: rm

File: src/Akeneo/Connectivity/Connection/back/tests/Acceptance/Context/ConnectionContext.php [hunk 6]
```diff
@@ -212,6 +222,16 @@ public function theConnectionImageShouldBe(string $label, string $expectedImage)
...
     }
 
+    /**
+     * @Then the Connection :label user role id should be :expectedUserRoleId
+     */
+    public function theConnectionUserRoleIdShouldBe(string $label, string $expectedUserRoleId): void
+    {
+        $connection = $this->connectionRepository->findOneByCode(self::slugify($label));
+
+        // Assert::eq($expectedUserRoleId, (string) );
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Acceptance/Context/ConnectionContext.php]: rm

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectConnectionWithCredentialsByCodeQuery.php [hunk 7]
```diff
@@ -26,26 +28,54 @@ public function __construct(DbalConnection $dbalConnection)
...
+    c.label,
+    c.flow_type,
+    c.image,
+    c.client_id,
+    client.random_id,
+    client.secret,
+    u.username,
+    urole.role_id,
+    g.id as group_id
+FROM akeneo_connectivity_connection c
```
Comment (pierrejolly) [inline src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectConnectionWithCredentialsByCodeQuery.php]: I'm not a huge fan of this kind of alias `c`, `u`...

File: src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/UpdateConnectionCommandSpec.php [hunk 8]
```diff
@@ -45,14 +52,27 @@ public function it_returns_null_if_there_is_no_image()
         $this->image()->shouldReturn(null);
     }
 
+
```
Comment (willy-ahva) [inline src/Akeneo/Connectivity/Connection/back/tests/Unit/spec/Application/Settings/Command/UpdateConnectionCommandSpec.php]: rm

File: src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectConnectionWithCredentialsByCodeQuery.php (hunk 7 above)
Comment (willy-ahva) [reply src/Akeneo/Connectivity/Connection/back/Infrastructure/Persistence/Dbal/Query/DbalSelectConnectionWithCredentialsByCodeQuery.php]: I totally agree! I strongly discourage this type of alias as it makes the request unreadable and unmaintainable.

**Pull request merged**